pkg/apis/core/validation/validation.go

```diff
@@ -4551,12 +4604,52 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 		return allErrs
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
```

Comment:
In general we don't check gates during validation.  This is to handle cases where we roll back versions with data already stored in the API - we don't want to let invalid data through an update or something.

Presumaby, if the gate is off and the feature is not in use, this will not throw a failure, right?

So do we really need this gatecheck?

Reply:
I think this was to avoid calling the rather heavyweight (imo) GetPodQOS function twice. Which reminds me, can we just dump all the code that calls GetPodQOS in favor of PodStatus.QOSClass which is an immutable value set during PodCreate with your KEP 2527 in mind. Is this a legitimate reliance on persisted values in PodStatus? (If yes, there are a few places this is used across the codebase that I want to toss out - so a separate PR)

Reply:
Yes, I think that's legitimate.

If that's the only purpose of the gate, please comment it.

Reply:
will do.